### PR TITLE
Upgrade Ruby to 2.5.7 and Rails 5.0

### PR DESCRIPTION
**Pull request by sarahsehr**
Heroku Platform Cedar-14 is deprecated and will be removed on May 1, 2020.  Prepare to switch to platform Heroku-18.

Ruby (MRI) 2.5.7 is the lowest supported ruby version for Heroku-18.  Rails also needs to be upgraded due to compatibility issues with the Ruby version.

https://devcenter.heroku.com/articles/ruby-support#ruby-versions
https://stackoverflow.com/questions/53213498/has-ruby-2-5-been-tested-with-rails-4-2

### CODE REVIEW

File: test/controllers/github_events_controller_test.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class GithubEventsControllerTest < ActionController::TestCase
 
     request.headers['HTTP_X_HUB_SIGNATURE'] = signature
 
-    post :payload, params: payload, session: { format: :json, number: 1, pull_request: { state: 'open' }, repository: { full_name: repository.name } }
+    post :payload, body: payload, params: { format: :json, number: 1, pull_request: { state: 'open' }, repository: { full_name: repository.name } }
```
Comment (pauldanielm) [inline test/controllers/github_events_controller_test.rb]: It seems like 1231564e8ac8711a7987fcba49f1005361443082 and this commit might be able to be combined, what was your reasoning for keeping them separate?

Comment (sarahsehr) [reply test/controllers/github_events_controller_test.rb]: No reason, I was just committing as I made progress.  I can squash.

File: lib/github_config.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ def initialize(application_id:, application_secret:, restrict_access_to_organiza
   def self.from_values(config_hash)
     new(application_id:                   config_hash["github_application"]["application_id"],
         application_secret:               config_hash["github_application"]["application_secret"],
-        restrict_access_to_organizations: config_hash["restrict_access_to_organizations"].split(','))
+        restrict_access_to_organizations: config_hash["restrict_access_to_organizations"]&.split(','))
```
Comment (pauldanielm) [inline lib/github_config.rb]: What is the purpose of this change?

File: test/controllers/github_events_controller_test.rb (hunk 1 above)
Comment (pauldanielm) [reply test/controllers/github_events_controller_test.rb]: Sure, I think it would be a good idea to squash these, but it's really minor.

File: lib/github_config.rb (hunk 2 above)
Comment (sarahsehr) [reply lib/github_config.rb]: `config_hash["restrict_access_to_organizations"]` is populated by the ENV variable `ALLOWED_ORGANIZATIONS`.  
This is the logic for determining access:

```
def organization_permitted?(organizations)
    if @restrict_access_to_organizations
      !(@restrict_access_to_organizations & organizations).empty?
    else
      true
    end
  end
```

So in order the machine test github user who is not in any org, the valueof `@restrict_access_to_organizations` needs to be falsy.  But it was impossible to get a falsy value that wouldn't throw an error when executing `config_hash["restrict_access_to_organizations"].split(','))`.  I spent a lot of time trying to find an ENV variable format that would parse and set it the way I wanted, but I had no choice but to change the code to make this work.

File: test/controllers/github_events_controller_test.rb (hunk 1 above)
Comment (sarahsehr) [reply test/controllers/github_events_controller_test.rb]: Fixed: 1297d8bfdd257f40df2d36be3d6fccad31873bb4

File: lib/github_config.rb (hunk 2 above)
Comment (pauldanielm) [reply lib/github_config.rb]: I see, that sounds reasonable.
